From fa1b2fd77932da96693761d6eff39985104624f2 Mon Sep 17 00:00:00 2001
From: Alexander Mikhaylenko <alexm@gnome.org>
Date: Thu, 8 Oct 2020 12:18:35 +0500
Subject: [PATCH 1/2] carousel-box: Null-check child in get_page_at_position()

It can be null when there are no children in the carousel.
---
 src/hdy-carousel-box.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/hdy-carousel-box.c b/src/hdy-carousel-box.c
index 1e0355fb..cf6e6eb0 100644
--- a/src/hdy-carousel-box.c
+++ b/src/hdy-carousel-box.c
@@ -1723,7 +1723,7 @@ hdy_carousel_box_get_closest_snap_point (HdyCarouselBox *self)
  * the current position, the returned widget will match the currently
  * displayed page.
  *
- * Returns: the closest page.
+ * Returns: (nullable): the closest page.
  *
  * Since: 1.0
  */
@@ -1742,6 +1742,9 @@ hdy_carousel_box_get_page_at_position (HdyCarouselBox *self,
 
   child = get_closest_child_at (self, position, TRUE, FALSE);
 
+  if (!child)
+    return NULL;
+
   return child->widget;
 }
 
-- 
GitLab


From 637e55d410d856d64b4dc57b895b585aa8dfd068 Mon Sep 17 00:00:00 2001
From: Alexander Mikhaylenko <alexm@gnome.org>
Date: Fri, 9 Oct 2020 13:13:23 +0500
Subject: [PATCH 2/2] carousel-box: Don't invalidate cache on any allocation

Welp.

Fixes https://gitlab.gnome.org/GNOME/libhandy/-/issues/304
---
 src/hdy-carousel-box.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/hdy-carousel-box.c b/src/hdy-carousel-box.c
index cf6e6eb0..0c4c5759 100644
--- a/src/hdy-carousel-box.c
+++ b/src/hdy-carousel-box.c
@@ -903,7 +903,6 @@ hdy_carousel_box_size_allocate (GtkWidget     *widget,
     gtk_widget_size_allocate (child, &alloc);
   }
 
-  invalidate_drawing_cache (self);
   gtk_widget_set_clip (widget, allocation);
 }
 
-- 
GitLab

